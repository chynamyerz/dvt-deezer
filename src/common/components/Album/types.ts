import { IAlbum } from "../../types";

export interface AlbumProps {
  data: IAlbum;
}
