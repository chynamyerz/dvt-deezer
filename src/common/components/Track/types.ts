import { ITrack } from "../../types";

export interface TrackProps {
  data: ITrack;
}
