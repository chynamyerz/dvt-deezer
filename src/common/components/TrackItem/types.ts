import { ITrack } from "../../types";

export interface TrackItemProps {
  data: ITrack;
  index: number;
}
