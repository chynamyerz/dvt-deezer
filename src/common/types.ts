export interface IArtist {
  id: string | number;
  name: string;
  link?: string;
  share?: string;
  picture?: string;
  picture_small?: string;
  picture_medium?: string;
  picture_big?: string;
  picture_xl?: string;
  radio?: true;
  tracklist: string;
  nb_fan?: number;
  type: string;
}

export interface IContributor extends Omit<IArtist, "nb_fan"> {
  role: string;
}

export interface IGenre {
  id: number;
  name: string;
  picture: string;
  type: string;
}

// id: "302127",
// title: "Discovery",
// cover: "https://api.deezer.com/album/302127/image",
// cover_small:
//   "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/56x56-000000-80-0-0.jpg",
// cover_medium:
//   "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/250x250-000000-80-0-0.jpg",
// cover_big:
//   "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/500x500-000000-80-0-0.jpg",
// cover_xl:
//   "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/1000x1000-000000-80-0-0.jpg",
// md5_image: "2e018122cb56986277102d2041a592c8",
// tracklist: "https://api.deezer.com/album/302127/tracks",
// type: "album",
export interface IAlbum {
  id: string;
  title: string;
  upc?: string;
  link?: string;
  share?: string;
  cover: string;
  cover_small: string;
  cover_medium: string;
  cover_big: string;
  cover_xl: string;
  md5_image: string;
  genre_id?: 113;
  genres?: {
    data: IGenre[];
  };
  label?: string;
  nb_tracks?: number;
  duration?: number;
  fans?: number;
  release_date?: string;
  record_type?: string;
  available?: boolean;
  tracklist: string;
  explicit_lyrics?: boolean;
  explicit_content_lyrics?: number;
  explicit_content_cover?: number;
  contributors?: IContributor[];
  artist?: IArtist;
  type: string;
  tracks: {
    data: ITrack[];
  };
}

export interface ITrack {
  id: string;
  readable: boolean;
  title: string;
  title_short: string;
  title_version: string;
  isrc: string;
  link: string;
  share: string;
  duration: string;
  track_position: number;
  disk_number: number;
  rank: string;
  release_date: string;
  explicit_lyrics: boolean;
  explicit_content_lyrics: number;
  explicit_content_cover: number;
  preview: string;
  bpm: number;
  gain: number;
  available_countries: string[];
  contributors: IContributor[];
  md5_image: string;
  artist: IArtist;
  album: IAlbum;
  type: string;
}
