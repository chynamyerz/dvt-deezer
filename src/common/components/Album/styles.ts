import { makeStyles } from "@mui/styles";

export const useStyles = makeStyles(() => {
  return {
    title: {
      fontSize: 18,
      fontWeight: "bold",
      paddingBottom: 10,
    },
    releaseDate: {
      color: "gray",
      paddingBottom: 10,
    },
    textOverflow: {
      whiteSpace: "nowrap",
      overflow: "hidden",
      textOverflow: "ellipsis",
      width: "100%",
    },
  };
});
