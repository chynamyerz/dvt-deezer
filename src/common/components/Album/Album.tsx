import React, { FC } from "react";
import { Card, CardContent, CardMedia, Grid } from "@mui/material";
import clsx from "clsx";

import { AlbumProps } from "./types";
import { useStyles } from "./styles";
import moment from "moment";

export const Album: FC<AlbumProps> = ({
  data: { title, cover_big, release_date },
}) => {
  const classes = useStyles();

  return (
    <Card>
      <CardMedia component="img" height="200" src={cover_big} alt={title} />
      <CardContent>
        <Grid container direction="column">
          <Grid className={clsx(classes.title, classes.textOverflow)} item>
            {title}
          </Grid>

          <Grid
            className={clsx(classes.releaseDate, classes.textOverflow)}
            item
          >
            {moment(release_date).year()}
          </Grid>
        </Grid>
      </CardContent>
    </Card>
  );
};
