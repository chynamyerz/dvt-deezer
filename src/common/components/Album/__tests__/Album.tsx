import React from "react";
import { screen, render, RenderResult } from "@testing-library/react";

import { IAlbum } from "../../../types";
import { Album } from "../Album";
const mockData: IAlbum = {
  id: "302127",
  title: "Discovery",
  upc: "724384960650",
  link: "https://www.deezer.com/album/302127",
  share:
    "https://www.deezer.com/album/302127?utm_source=deezer&utm_content=album-302127&utm_term=0_1662356989&utm_medium=web",
  cover: "https://api.deezer.com/album/302127/image",
  cover_small:
    "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/56x56-000000-80-0-0.jpg",
  cover_medium:
    "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/250x250-000000-80-0-0.jpg",
  cover_big:
    "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/500x500-000000-80-0-0.jpg",
  cover_xl:
    "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/1000x1000-000000-80-0-0.jpg",
  md5_image: "2e018122cb56986277102d2041a592c8",
  genre_id: 113,
  genres: {
    data: [
      {
        id: 113,
        name: "Dance",
        picture: "https://api.deezer.com/genre/113/image",
        type: "genre",
      },
    ],
  },
  label: "Daft Life Ltd./ADA France",
  nb_tracks: 14,
  duration: 3660,
  fans: 262009,
  release_date: "2001-03-07",
  record_type: "album",
  available: true,
  tracklist: "https://api.deezer.com/album/302127/tracks",
  explicit_lyrics: false,
  explicit_content_lyrics: 7,
  explicit_content_cover: 0,
  contributors: [
    {
      id: 27,
      name: "Daft Punk",
      link: "https://www.deezer.com/artist/27",
      share:
        "https://www.deezer.com/artist/27?utm_source=deezer&utm_content=artist-27&utm_term=0_1662356989&utm_medium=web",
      picture: "https://api.deezer.com/artist/27/image",
      picture_small:
        "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/56x56-000000-80-0-0.jpg",
      picture_medium:
        "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/250x250-000000-80-0-0.jpg",
      picture_big:
        "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/500x500-000000-80-0-0.jpg",
      picture_xl:
        "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/1000x1000-000000-80-0-0.jpg",
      radio: true,
      tracklist: "https://api.deezer.com/artist/27/top?limit=50",
      type: "artist",
      role: "Main",
    },
  ],
  artist: {
    id: "27",
    name: "Daft Punk",
    picture: "https://api.deezer.com/artist/27/image",
    picture_small:
      "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/56x56-000000-80-0-0.jpg",
    picture_medium:
      "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/250x250-000000-80-0-0.jpg",
    picture_big:
      "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/500x500-000000-80-0-0.jpg",
    picture_xl:
      "https://e-cdns-images.dzcdn.net/images/artist/f2bc007e9133c946ac3c3907ddc5d2ea/1000x1000-000000-80-0-0.jpg",
    tracklist: "https://api.deezer.com/artist/27/top?limit=50",
    type: "artist",
  },
  type: "album",
  tracks: {
    data: [
      {
        id: "3135553",
        readable: true,
        title: "One More Time",
        title_short: "One More Time",
        title_version: "",
        link: "https://www.deezer.com/track/3135553",
        duration: "320",
        rank: "902876",
        explicit_lyrics: false,
        explicit_content_lyrics: 0,
        explicit_content_cover: 0,
        preview:
          "https://cdns-preview-e.dzcdn.net/stream/c-e77d23e0c8ed7567a507a6d1b6a9ca1b-9.mp3",
        md5_image: "2e018122cb56986277102d2041a592c8",
        artist: {
          id: "27",
          name: "Daft Punk",
          tracklist: "https://api.deezer.com/artist/27/top?limit=50",
          type: "artist",
        },
        //@ts-ignore
        album: {
          id: "302127",
          title: "Discovery",
          cover: "https://api.deezer.com/album/302127/image",
          cover_small:
            "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/56x56-000000-80-0-0.jpg",
          cover_medium:
            "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/250x250-000000-80-0-0.jpg",
          cover_big:
            "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/500x500-000000-80-0-0.jpg",
          cover_xl:
            "https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/1000x1000-000000-80-0-0.jpg",
          md5_image: "2e018122cb56986277102d2041a592c8",
          tracklist: "https://api.deezer.com/album/302127/tracks",
          type: "album",
        },
        type: "track",
      },
    ],
  },
};

let component: RenderResult | null = null;

beforeEach(() => {
  component = render(<Album data={mockData} />);
});

describe("<Album>", () => {
  it("renders the component", () => {
    expect(component?.asFragment()).toMatchSnapshot();
  });

  it("has correct name", () => {
    expect(screen.queryByText("Discovery")).toBeInTheDocument();
  });
  it("has correct year", () => {
    expect(screen.queryByText("2001")).toBeInTheDocument();
  });
});
