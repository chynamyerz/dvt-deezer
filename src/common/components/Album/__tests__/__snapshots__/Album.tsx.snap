// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Album> renders the component 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Discovery"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="200"
      src="https://e-cdns-images.dzcdn.net/images/cover/2e018122cb56986277102d2041a592c8/500x500-000000-80-0-0.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item makeStyles-title-1 makeStyles-textOverflow-3 css-13i4rnv-MuiGrid-root"
        >
          Discovery
        </div>
        <div
          class="MuiGrid-root MuiGrid-item makeStyles-releaseDate-2 makeStyles-textOverflow-3 css-13i4rnv-MuiGrid-root"
        >
          2001
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
